lib/wrapThenable.js

```diff
@@ -5,12 +5,17 @@ const {
   kReplyHijacked
 } = require('./symbols')
 
-function wrapThenable (thenable, reply) {
+function wrapThenable (thenable, reply, channels, store) {
   thenable.then(function (payload) {
     if (reply[kReplyHijacked] === true) {
       return
     }
 
+    if (channels) {
+      store.result = payload
+      channels.asyncStart.publish(store)
+    }
```

Comment:
I'm not 100% convinced of the meaning of asyncStart. Can you clarify why you put it here?

Reply:
With callbacks there's asyncStart and asyncEnd scoping the callback. Most tracers want to create a span up to the start of the callback, but some consider the callback to be part of the task and so want to capture up to the end. In the case of promises there's no distinct "end" condition other than GC so it generally emits the two together for promises. I think the logic here was that it's surrounding the _internal_ continuation of the thenable rather than just have the two events right beside each other without scoping _anything_ due to that limitation of not knowing the _actual_ end. Does that makes sense here?